rust/src/encodings/binary.rs

```diff
@@ -230,9 +231,22 @@ impl<'a, T: ByteArrayType> BinaryDecoder<'a, T> {
                 .null_bit_buffer(Some(null_buf.into()));
         }
 
+        // TODO: replace this with safe method once arrow-rs 47.0.0 comes out.
+        // https://github.com/lancedb/lance/issues/1237
+        // Zero-copy conversion from bytes
+        // Safety: the bytes are owned by the `data` value, so the pointer
+        // will be valid for the lifetime of the Arc we are passing in.
+        let buf = unsafe {
```

Comment:
we don't need the alignment check because this is already byte_width aligned?
also, technically `bytes` can be empty right? Would that cause any issues with `NonNull::new(bytes.as_ptr() as _).unwrap()`?

Reply:
Yeah for bytes I think we just require need it to be aligned with u8, which is true of any allocation.

Reply:
Our simd code uses unaligned load now, so it is unnecessary to be word-aligned (i.e., 8 bytes). About 10% overhead due to misaligned memory loading in SIMD computation, much less in overall e2e latency.

Reply:
Yeah I align the primitive buffers because arrow-rs errors otherwise